utils/memset_safe.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2019 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "utils/memset_safe.h"
+#include <stdint.h>
+
+#ifndef __STDC_LIB_EXT1__
+
+retcode_t memset_safe(void *dest, size_t destsz, int ch, size_t count) {
+  if (dest == NULL) {
+    return RC_OOM;
+  } else if (destsz > SIZE_MAX) {
+    return RC_OOM;
+  } else if (count > destsz) {
+    return RC_OOM;
+  }
```

Comment:
It doesn't make sense to return `RC_OOM` (Out Of Memory) here.
We shoule return errors that are similar to the real `memset_s`.
```
     [EINVAL]           The s argument was a null pointer.

     [E2BIG]            One or both of smax or n was larger than RSIZE_MAX.

     [EOVERFLOW]        n was larger than smax.
```